src/mscorlib/src/System/TypedReference.cs

```diff
@@ -42,14 +42,14 @@ public static TypedReference MakeTypedReference(Object target, FieldInfo[] flds)
                     throw new ArgumentException(SR.Argument_MustBeRuntimeFieldInfo);
 
                 if (field.IsInitOnly || field.IsStatic)
-                    throw new ArgumentException(SR.Argument_TypedReferenceInvalidField);
+                    throw new ArgumentException(SR.Format(SR.Argument_TypedReferenceInvalidField, field.Name));
```

Comment:
@danmosemsft do you think is better use paramName [name overloads](https://msdn.microsoft.com/en-us/library/sxykka64(v=vs.110).aspx)?I don't know if is correct...the real method parameter is fields array, but also every element of this array could an "argument", do you know if there are other similar context where take a cue?

Reply:
You're probably right that we should only use that if the value is actually a parameter name from the user's point of view.